lib/resty/openssl/pkey.lua

```diff
@@ -1060,7 +1060,12 @@ function _M.paramgen(config)
     return nil, format_error("pkey.paramgen: EVP_PKEY_get0_{key}")
   end
 
-  return bio_util.read_wrap(write_func, ctx)
+  local res, err = bio_util.read_wrap(write_func, ctx)
+  params = nil
+  return res, err
+  -- use the following form the params would be gc colloected
+  -- before the bio is read
+  -- return bio_util.read_wrap(write_func, ctx)
```

Comment:
👍

Reply:
After a second thought, I think param is not used after get_ctx_func returns, so there's no use-after-free here.
On the other side, as we already set the GC handler when returning from `generate_param`, the `params` could have been GC'ed after line 1058, so this fix doesn't feel correct to me.